src/proto/network/snp.rs

```diff
@@ -0,0 +1,383 @@
+//! Network I/O protocols.
+
+use crate::proto::Protocol;
+use crate::{unsafe_guid, Result, Status};
+
+/// The Snp protocol.
+#[repr(C)]
+#[unsafe_guid("a19832b9-ac25-11d3-9a2d-0090273fc14d")]
+#[derive(Protocol)]
+pub struct Snp {
```

Comment:
Some other protocols in the crate with "simple" in their name ended up with the noun part of the protocol name (e.g. `SimpleTextOutput` became just `Output`, `GraphicsOutputProtocol` became `GraphicsOutput`). What do you say about renaming this to something like `Network`?

Reply:
What if someone wants to add the chapter 25 Managed Network Protocol (MNP) later on? Maybe it would be called `ManagedNetwork`?

That's fine with me. I'll re-spin this PR pretty soon with the new name and a bunch of other improvements.